src/infra/core/src/ingest/push_ingest_service_impl.rs

```diff
@@ -84,19 +84,25 @@ impl PushIngestServiceImpl {
         let ctx: SessionContext =
             ingest_common::new_session_context(self.object_store_registry.clone());
 
-        let data_writer = match DataWriterDataFusion::builder(dataset.clone(), ctx.clone())
-            .with_metadata_state_scanned(source_name)
-            .await
-        {
-            Ok(b) => Ok(b.build()),
-            Err(ScanMetadataError::SourceNotFound(err)) => {
-                Err(PushIngestError::SourceNotFound(err.into()))
-            }
-            Err(ScanMetadataError::Internal(err)) => Err(PushIngestError::Internal(err)),
-        }?;
+        let mut data_writer = self
+            .make_data_writer(dataset.clone(), source_name, ctx.clone())
+            .await?;
 
         let push_source = match data_writer.source_event() {
+            None => {
+                let add_push_source_event = self
+                    .auto_create_push_source(dataset.clone(), "auto", &opts)
```

Comment:
Can we add an `auto_create_push_source` parameter to `PushIngestOpts` so we only auto-create push sources in file upload context. I wouldn't want such auto creation happening any time someone calls push API.

Reply:
Added, I agree this is safer.